lib/app/service/sync.go

```diff
@@ -133,6 +136,13 @@ func SyncApp(ctx context.Context, req SyncRequest) error {
 
 	log.Infof("Syncing %v.", req.Package)
 
+	if req.ScanConfig != nil {
+		if err = req.ImageService.SyncForScanning(ctx, *req.ScanConfig, syncPath, req.Progress); err != nil {
```

Comment:
I think it would be cleaner if you had a separate implementation of ImageService that would reimplement "Sync" method so it would properly sync images for scanning, and just pass it here from the caller - then this method won't need any updating at all, and no new ScanConfig parameters would be needed, so the SyncApp would be completely agnostic to what kind of syncing it should do and just use the passed in image service.